### ISSUE TITLE
Exporting a data.frame with Variables' values at presence and (pseudo)absences locations

### ISSUE THREAD

**Issue by LorenzoBernicchi** [biomodhub/biomod2#246]
Hello everyone, 
I am developing a pretty simple species distribution model for a species in Italy. I did not encounter any particular error during my modeling, but I have a question to ask about the possible presence of a specific feature; if the question is off-topic, I apologize in advance.
I would like to create a `data.frame` in which are stored the values of the variables extracted at both the presence and (pseudo)absence locations. 
I know it would be pretty easy to do that for my presences, but since my (pseudo)absences are created within the `BIOMOD_FormatingData()` I do not know how to export it and create what I am looking for.

I really hope you can help,
I wish you a great day!

**Comment by rpatin** [comment]
Hello @LorenzoBernicchi 
Thank you for reporting :pray:
It is perfectly fine to ask such question here. We want this issue pages to be both a place where people can report bugs, ask questions and also seek help for their modelling with `biomod2`.

You can use [`get_formal_data`](https://biomodhub.github.io/biomod2/reference/getters.out.html) to retrieve the information you want.
With a model output `myBiomodModelOut` obtained from `BIOMOD_Modeling` but also with your formated data `myBiomodData` you can get everything.
With
```
get_formal_data(myBiomodModelOut, subinfo = "resp.var")
# or 
myBiomodData@data.species
```
You get the response variable as a vector (0/1 for presence/absence ; NA/1 for presence/pseudo-absence)
and with 
```
get_formal_data(myBiomodModelOut, subinfo = "expl.var"))
# or 
myBiomodData@data.env.var
```
you can get the extracted environmental variables.
You can also get the coordinates of your points if any:
```
get_formal_data(myBiomodModelOut)@coord
# or 
myBiomodData@coord
```
In summary to generate the `data.frame` you are looking for:
```
full.data <- 
  cbind(data.frame("presence" = get_formal_data(myBiomodModelOut, subinfo = "resp.var")),
        get_formal_data(myBiomodModelOut)@coord,
        get_formal_data(myBiomodModelOut, subinfo = "expl.var"))
```

I hope this answer your question. If not, feel free to update the issue with more details !
Have a great day,
Rémi

**Comment by rpatin** [comment]
Thanks @MayaGueguen that just reminded me of new function recently added: `get_species_data`, that does everything that you need with much more simple code :tada:  
```
get_species_data(myBiomodData)
```
or 
```
myBiomodData <- get_formal_data(myBiomodModelOut)
get_species_data(myBiomodData)
```

Note that many information of this kind can still be accessed with `get_formal_data` which is still a useful function to know.

**Comment by LorenzoBernicchi** [comment]
Thank you very much @rpatin and @MayaGueguen for your fast and really precise answer, that helped a lot! 
I wish you a great day, thanks again!
